File: src/main/java/com/sunshines/bookstore/Repository/UserRepository.java
package com.sunshines.bookstore.Repository;

import com.sunshines.bookstore.Model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserRepository extends JpaRepository<User, Integer> {
    List<User> findAll();
    List<User> findAllByEmail(String email);

    User findFirstByEmail(String email);
    User findFirstById(int id);
}

File: src/main/java/com/sunshines/classrooms/Repository/UserRepository.kt
package com.sunshines.classrooms.Repository

import com.sunshines.classrooms.Model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Int> {
    override fun findAll(): List<User>
    fun findAllByEmail(email: String): List<User>

    fun findFirstByEmail(email: String): User
    fun findFirstById(id: Int): User
}
